File: python_roguelike/data/pools/effect_pool.py
from typing import Dict, Optional
from ..effect_data import EffectData


class EffectPool:
    def __init__(self):
        self.effects_by_id: Dict[str, EffectData] = {}

    def get_effect(self, id: str) -> Optional[EffectData]:
        return self.effects_by_id.get(id)

File: src/Roguelike/Core/Data/DataPool/EffectPool.cs
using System.Collections.Generic;

namespace Roguelike.Data
{


    public class EffectPool
    {
        public Dictionary<string, EffectData> EffectsById { get; set; } = new Dictionary<string, EffectData>();

        public EffectData GetEffect(string id)
        {
            EffectsById.TryGetValue(id, out var effect);
            return effect;
        }
    }
}
